src/webgpu/api/validation/state/device_lost/destroy.spec.ts

```diff
@@ -0,0 +1,184 @@
+export const description = `Validation tests for device lost induced via destroy.`;
+
+import { makeTestGroup } from '../../../../../common/framework/test_group.js';
+import {
+  kBufferUsageTypes,
+  kBufferUsageTypesInfo,
+  kBufferUsageCopy,
+  kBufferUsageCopyInfo,
+  kTextureUsageType,
+  kTextureUsageTypeInfo,
+  kTextureUsageCopy,
+  kTextureUsageCopyInfo,
+  kRegularTextureFormats,
+  kCompressedTextureFormats,
+  kTextureFormatInfo,
+} from '../../../../capability_info.js';
+import { ValidationTest } from '../../validation_test.js';
+
+export const g = makeTestGroup(ValidationTest);
+
```

Comment:
how did we decide what resources to test creation of?
if there's other stuff to add testing for, we need to plan it out with TODOs or no one will know what remains.

Reply:
Ah... I added more tests for now. The plan is to test all the API. I decided I will just implement them all in this PR, so it will be a WIP. Feel free to make suggestions as I add commits, or once at the end if you prefer. Converting the PR to a draft for now.